docs/content/guides/6.multistore/2.tooling-and-concepts/3.development/4.local-environment.md

```diff
@@ -7,6 +7,187 @@ navigation:
+The applications will run on ports specified in the `alokai.config.json` file. For example:
+
+```json
+{
+  "stores": {
+    "default": {
+      "localPorts": {
+        "middleware": 4000,
+        "nextjs": 3000,
+        "nuxt": 3333
+      }
+    }
+  }
+}
+```
+
+Each store can have its own port configuration to avoid conflicts when running multiple stores simultaneously.
+::
+
+### Using Local Domains
+
+Alokai uses Caddy as a reverse proxy to handle local domains. The Caddyfile is automatically generated based on the `alokai.config.json` file and the `localDomain` property for each store.
+
+#### Prerequisites
+
+- [Caddy server installed](https://caddyserver.com/docs/install)
+- Local domains configured in your hosts file. For example:
+  - If you have a store with `localDomain` set to `default.local`, add `127.0.0.1 default.local` to your hosts file
+
+#### Starting with local domains
+
+Start all stores with their local domains:
+
+```bash
+yarn store dev --all --with-local-domains
+```
+
+:::tip HTTPS
+Caddy automatically generates self-signed SSL certificates for local domains. You 
+can read more about it in the [Caddy documentation](https://caddyserver.com/docs/automatic-https).
+:::
+
+## Building for Production
+
+To test the production build locally:
+
+1. Build all deployable stores:
+```bash
+yarn build
+```
+
+2. Start the production server:
+```bash
+yarn start
+```
+
+Note that template stores are not built into the `.out` directory, as they only serve as templates for their descendant stores.
+
+## Running Integration Tests
+
+Integration tests run against the composed stores, respecting the file inheritance chain:
+
+```bash
+yarn test:integration:pw
+```
+
+You can also run tests in Playwright's UI mode for better debugging experience:
+
+```bash
+yarn store test --store-id=my-store --ui 
+```
+
+:::warning
+The `--ui` flag only works when testing a single store. Make sure to specify the store using the `--store-id` flag.
+:::
+
+For more information on integration tests, refer to the [Integration Tests](/guides/multistore/tooling-and-concepts/development/integration-tests) guide.
+
+## Common Issues and Solutions
+
+### Cache Problems
+
+If you experience unexpected behavior with file inheritance or turbo cache. For example:
+- You modify a component in your store, but the changes are not reflected in the browser
+- The `.out` directory still contains old versions of your files after making changes
+
+Here's how to fix it:
+
+```bash
+# Clear Turbo cache
+rm -rf .turbo/cache
+
+# Clear composed store outputs
+rm -rf .out/*
+
+# Rebuild all stores
+yarn build
+```
+
+::alert{type="warning"}
+The `.out` directory contains the composed stores and is in `.gitignore`. If you experience inheritance issues, try clearing this directory and rebuilding.
+::
+
+## Development Best Practices
+
+1. **Use Store-Specific Development**
+When working on a single store, use the `--store-id` flag to improve build times:
```

Comment:
```suggestion
When working on a single store, use the `--store-id` flag to improve build times. Otherwise, every time you make a change, all the stores will be rebuilt.
```

Reply:
In order to build all stores, the `--all` flag is required. It was introduced based on Rick's feedback. So if the user does not pass the `--store-id` or `--all` flag, the command will fail.